src/OpenTelemetry.ResourceDetectors.Container/README.md

```diff
@@ -34,6 +34,66 @@ your application is running:
 
 - **ContainerResourceDetector**: container.id.
 
+## Kubernetes
+
+When running in a Kubernetes environment, the Container resource detector
+requires permissions to access pod information in order
+to extract the container ID.
+This is achieved using Kubernetes Role-Based Access Control (RBAC).
+
+You will need to create a Role and RoleBinding to grant
+the necessary permissions to the service account used by your application.
+Below is an example of how to configure these RBAC resources:
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pod-reader-account
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: default
+  name: pod-reader
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: read-pods
+  namespace: default
+subjects:
+- kind: ServiceAccount
+  name: pod-reader-account
+  namespace: default
+roleRef:
+  kind: Role
+  name: pod-reader
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: container-resolve-demo
+spec:
+  serviceAccountName: pod-reader-account
+  volumes:
+  - name: shared-data
+    emptyDir: {}
+  containers:
+  - name: &container_name my_container_name
+    image: ubuntu:latest
+    command: [ "/bin/bash", "-c", "--" ]
+    args: [ "while true; do sleep 30; done;" ]
+    env:
+    - name: KUBERNETES_CONTAINER_NAME
+      value: *container_name
+```
```

Comment:
Should we add to readme how to define KUBERNETES_POD_NAME with downwards API?
```
    - name: KUBERNETES_POD_NAME
      valueFrom:
        fieldRef:
          fieldPath: metadata.name
```

Reply:
Done.